python_ta/typecheck/base.py

```diff
@@ -149,10 +149,50 @@ class TupleSubscript(TypeVar, _root=True):
     pass
 
 
-def create_Callable(args: Iterable[type], rtype, poly_vars=None):
-    poly_vars = poly_vars or []
+_TYPESHED_TVARS = {
+    '_T': TypeVar('_T'),
+    '_T_co': TypeVar('_T_co', covariant=True),
+    '_KT': TypeVar('_KT'),
+    '_VT': TypeVar('_VT'),
+    '_S': TypeVar('_S'),
+    '_T1': TypeVar('_T1'),
+    '_T2': TypeVar('_T2'),
+    '_T3': TypeVar('_T3'),
+    '_T4': TypeVar('_T4'),
+    '_T5': TypeVar('_T5'),
+    '_TT': TypeVar('_TT', bound='type'),
+    'function': Callable[[Any], Any]
+}
+
+
+_BUILTIN_TO_TYPING = {
+    'list': 'List',
+    'dict': 'Dict',
+    'tuple': 'Tuple',
+    'set': 'Set',
+    'frozenset': 'FrozenSet',
+    'function': 'Callable'
+}
+
+
+def _get_poly_vars(t: type):
+    if isinstance(t, TypeVar) and t.__name__ in _TYPESHED_TVARS:
+        return set([t.__name__])
+    elif isinstance(t, GenericMeta) and t.__args__:
+        pvars = set()
+        for arg in t.__args__:
+            pvars.update(_get_poly_vars(arg))
+        return pvars
+    return set()
+
+
+def create_Callable(args: Iterable[type], rtype, class_poly_vars=None):
+    poly_vars = class_poly_vars or []
+    poly_vars = set(poly_vars)
     c = Callable[list(args), rtype]
-    c.polymorphic_tvars = poly_vars
+    poly_vars.update(_get_poly_vars(c))
+    c.polymorphic_tvars = set()
+    c.polymorphic_tvars.update(poly_vars)
```

Comment:
This is an odd way to set `c.polymorphic_tvars` to `poly_vars`, but it is a workaround for a general problem with Callable attributes that we've been bumping into.
For example,
```
c = Callable[[int],int]
c.x = 3
c = Callable[[int],str]
print(c.x)  # prints 3
c.x = 2
print(c.x)  # prints 3
```